app/config/errors.php

```diff
@@ -195,6 +195,21 @@
         'description' => 'Missing ID from OAuth2 provider.',
         'code' => 400,
     ],
+    Exception::USER_OAUTH2_BAD_REQUEST => [
+        'name' => Exception::USER_OAUTH2_BAD_REQUEST,
+        'description' => 'OAuth2 provider rejected the bad request.',
```

Comment:
This is great!

I would still elaborate on these a little:
**[What went wrong] [Why we think it went wrong] [How you could fix it]**

OAuth2 provider rejected the bad request. We think [SOMETHING] is misconfigured. Check [PLACE] to see if there are any mistakes.

Reply:
@gewenyu99, these are the default descriptions shown in the docs, but they can be different from what's returned to the developer. For example, one error response we have is:

```json
{
    "message": "Failed to obtain access token. The Google OAuth2 provider returned an error: invalid_grant: Malformed auth code.",
    "type": "user_oauth2_bad_request",
    "code": 400
}
```

Reply:
Okay, would I be so bold to suggest saying: OAuth2 provider rejected the bad request. This error will be replaced by an error from the provider at runtime.

I know it's excessive, but maybe this is a good hint for maintainers, but also in docs.

I err on the side of over explaining with errors cause it's extra frustrating. You can make the call.

Reply:
> This error will be replaced by an error from the provider at runtime.

That doesn't really make sense to me because it's just a description of the error type.

Reply:
> > This error will be replaced by an error from the provider at runtime.
> 
> That doesn't really make sense to me because it's just a description of the error type.

:/ Maybe the real issue is how we present these errors. I never know which ones will have their description replaced at runtime, which ones will actually use the description in the error file.

I'll let it slide, cause it's not really developer facing, more a maintainer facing issue.